src/poliastro/core/spheroid_location.py

```diff
@@ -177,11 +177,13 @@ def cartesian_to_ellipsoidal(_a, _c, x, y, z):
     e2_ = e2 / (1 - e2)
     p = np.sqrt(x ** 2 + y ** 2)
     th = np.arctan(z * _a / (p * _c))
-    lon = np.arctan(y / x)
+    lon = np.arctan2(y, x)  # Use `arctan2` so that lon lies in the range: [-pi, +pi]
     lat = np.arctan((z + e2_ * _c * np.sin(th) ** 3) / (p - e2 * _a * np.cos(th) ** 3))
 
     v = _a / np.sqrt(1 - e2 * np.sin(lat) ** 2)
-    h = p / np.cos(lat) - v
-    h = z / np.sin(lat) - (1 - e2) * v
+    if lat == np.pi / 2 or lat:
+        h = z / np.sin(lat) - (1 - e2) * v
```

Comment:
Will this calculation fail for values of latitude _close_ to the singularity? Or only if it ends up being 0? I propose we add tests with hypothesis to check

Reply:
I couldn't generate an example where this fails for near-zero latitude values, but the `ZeroDivisionError` that I had got, would mostly only raise if the latitude is exactly zero.

Also, I am not quite sure of the added hypothesis test since I am not yet familiar with the `hypothesis` library. I would need to work on it too.

Reply:
The added test would check for `lat` values between -0.01 rad and +0.01 rad. For example, it checks for:

<details>

```
-0.0 rad
-0.002507834041561746 rad
0.001713937132179604 rad
0.003418967494897612 rad
-0.005041564477248276 rad
0.00010227531601860964 rad
0.005457346579813051 rad
0.0004731085445776186 rad
-0.0023325910925268596 rad
0.007986193709709937 rad
-0.004545211335889764 rad
0.002982056394020967 rad
5.07483729500713e-06 rad
0.0012991583475218252 rad
0.0004685299038809033 rad
0.009943655393511225 rad
0.0002089376333545157 rad
0.00018938777605675755 rad
-0.002471235273631139 rad
-0.008406899317204104 rad
0.009337363317096326 rad
0.00042561325525483315 rad
0.0002991170674155753 rad
0.006573969258387252 rad
0.00021421357362847452 rad
0.004838674848889469 rad
0.00038799643998632454 rad
0.006391198303459386 rad
0.00047169852352819914 rad
0.0001521530074741007 rad
0.00895116991336977 rad
0.0014994978226605185 rad
0.0038714426010926727 rad
0.000276203342353334 rad
0.000779733675969725 rad
0.0004999123889504965 rad
0.0005443053293808787 rad
0.0005526081774104609 rad
0.0014676934170779535 rad
0.0003165570492322756 rad
0.000191077173685319 rad
-0.001678972410652393 rad
0.001068676455303608 rad
-0.003581172557723565 rad
-0.0065418170714943834 rad
0.0037242919138848925 rad
0.0003298332198871768 rad
0.0044373042911172444 rad
0.000368484996232823 rad
0.004332159035602673 rad
0.009032713114283995 rad
0.002374557256702051 rad
-0.0017038910111076592 rad
0.006372009452072477 rad
0.0012344197305538044 rad
-0.006011451021773841 rad
-0.00429656455029835 rad
-0.008444957184234008 rad
0.009310300456300397 rad
0.0034369168129012766 rad
-0.00985070410272671 rad
0.008334682811651579 rad
0.0005201680576278945 rad
0.003163022752740972 rad
-0.009733824701688732 rad
0.0011084857069979818 rad
0.00047890413733516114 rad
0.002599459157801221 rad
0.005461544397112458 rad
0.00030551095830686056 rad
0.008210805326556285 rad
0.00040431583705091044 rad
0.004673496085552767 rad
-0.00641499790150819 rad
0.007405984171696089 rad
-0.005931947954198361 rad
-0.0018501871962451726 rad
0.008193226385198649 rad
0.0003056303129001848 rad
0.00028029436872252907 rad
0.0017553583929674236 rad
0.009371748599660357 rad
0.0004576727168910356 rad
0.00716421552410509 rad
-0.008451956407481864 rad
0.0028279642863069526 rad
0.00395885729457974 rad
0.0034674674124130574 rad
-0.0076716575777425215 rad
0.009402836007267443 rad
0.0004185623653217619 rad
0.00715196552237103 rad
-0.0009031737269833929 rad
0.0010925372631223733 rad
-0.009689539359327459 rad
0.003879099085201888 rad
0.003049365811683074 rad
0.0006376477908668311 rad
0.00034565610379164955 rad
```

</details>

Would this be enough proof for validating the computation for near-zero latitudes?